index.js

```diff
@@ -37,26 +56,44 @@ function validateEPSG (srid) {
  * @param  {string} info The projection info being requested
  * @return {function} callback Returns projection info requested
  */
-Ptolemy.get = function(epsg, format, callback) {
-  epsg = epsg.toString();
-  if (validateEPSG(epsg)) {
-    if (formatWhiteList.indexOf(format) > -1) {
-      var requestURI = BASE_URI + epsg + '.' + format;
-      needle.get(requestURI, {timeout:4000}, function(error, response) {
-        if (!error && response && response.statusCode == 200){
-          callback(null, response.body);
-        } else if (response && response.statusCode) {
-          callback(response.statusCode + ' : ' + response.body);
-        } else {
-          callback(error);
-        }
-      });
-    } else {
-      callback('Please make sure you are requesting one of the supported formats.');
-    }
-  } else {
-    callback('Please make sure you are providing a valid EPSG SRID');
+Ptolemy.prototype.get = function (epsg, format) {
+	epsg = epsg.toString();
+  var returnObj = {};
+  returnObj.epsg = epsg;
+
+  if (!isValidEPSG(epsg)) {
+    return Bluebird.reject(new InvalidSRIDError('Invalid EPSG SRID.'));
+  }
+  
+  if (!(formatWhiteList.indexOf(format) > -1)) {
+    return Bluebird.reject(new InvalidFormatError('Invalid format.'));
   }
-};
+	
+	var requestURL = BASE_URL + epsg + '.' + format;
+  // Note: We parse the SRID's XML to get the Site Name (avoids scraping)
+	var nameURL = BASE_URL + epsg + '.xml';
+  
+	return promisfiedGet(requestURL, {timeout: 4000})
+	.then((res) => {
+    if (res.statusCode != 200) {
+      return Bluebird.reject(new Error(res.statusCode));
```

Comment:
Create a new error and add the response body. Status code alone is pretty useless.